src/main/java/com/googleinterns/zoomtube/servlets/TranscriptServlet.java

```diff
@@ -63,18 +83,52 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       // A for loop is used because NodeList is not an Iterable.
       for (int nodeIndex = 0; nodeIndex < nodeList.getLength(); nodeIndex++) {
         Node node = nodeList.item(nodeIndex);
-        Element element = (Element) node;
-        String lineStart = element.getAttribute(START_ATTRIBUTE);
-        String lineDuration = element.getAttribute(DURATION_ATTRIBUTE);
-        String lineContent = node.getTextContent();
-
-        // TODO: Remove print statement. It is currently here for display purposes.
-        System.out.println(lineStart + " " + lineDuration + " "
-            + " " + lineContent);
+        createEntity(node, lectureId);
       }
     } catch (ParserConfigurationException | SAXException e) {
       // TODO: alert the user.
       System.out.println("XML parsing error");
     }
   }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long lectureId = Long.parseLong(request.getParameter(PARAM_LECTURE_ID));
+    Key lecture = KeyFactory.createKey(PARAM_LECTURE, lectureId);
+
+    Filter lectureFilter = new FilterPredicate(Line.PROP_LECTURE, FilterOperator.EQUAL, lecture);
+
+    Query query = new Query(Line.ENTITY_KIND)
+                      .setFilter(lectureFilter)
+                      .addSort(Line.PROP_START, SortDirection.ASCENDING);
+    PreparedQuery pq = datastore.prepare(query);
+
+    ImmutableList.Builder<Line> lineBuilder = new ImmutableList.Builder<>();
+    for (Entity entity : pq.asQueryResultIterable()) {
+      lineBuilder.add(Line.fromEntity(entity));
+    }
+    ImmutableList<Line> lines = lineBuilder.build();
+
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(lines));
+  }
+
+  /**
+   * Creates an entity using the attributes from {@code node} and {@code lectureId}.
+   */
+  private void createEntity(Node node, long lectureId) {
```

Comment:
Can this be changed to createLineEntity? or include what `kind` of entity is being made

Reply:
I agree